### DISCUSSION TITLE
autostart feature

### DISCUSSION THREAD

**Discussion by bocianu** [paulscottrobson/neo6502-firmware#267] [Implemented Ideas]
When the system at startup finds a file with a predefined name (e.g. autorun.neo), it launches it immediately instead of basic prompt.
I think such a feature could be a nice addition. 

I'm just wondering what type of file it should be, a regular binary or maybe a basic file (if possible)? 
You can load a binary from basic, and basic also allows you to create a simple menu, what should it run, so it gives you more possibilities.

what do you think about it?

**Comment by paulscottrobson** [comment]
I've always wanted something like autoexec.bat, !boot or whatever. The issue with it is about having a way to skip it, if you want to change it or make a mistake. The usual way of doing this is something like holding down shift on startup. I don't think you can check the status of shift on power up (might be possible) because that is sent with the Keyboard packets from USB when you press a key. I will experiment :)

**Comment by jviskari** [comment]
Well, let's not over engineer this.  If Morpheus just loads and runs the start up script, say 'matrix_enter.bas',  from the root of the mass storage that should cover majority of use cases.   If the file does not exists, then it can be skipped. 

If the basic script is faulty, user most likely gets a syntax error. If the script somehow gets stuck, this can be fixed simply by removing USB/SD card before powering on Neo6502. 

I would most likely use this kind of autostart script to set locale, customize palette and configure f-keys.
